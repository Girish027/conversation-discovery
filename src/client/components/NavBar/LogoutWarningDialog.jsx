import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { Dialog, KeyboardUtils, StringUtils } from '@tfs/ui-components';
import { styles } from '../../styles/customComponentsStyles';
import constants from '../../constants';

const Card = ({ iconLink, title }) => (
  <div
    role='presentation'
    style={styles.card}
  >
    <img src={iconLink} alt='Application logo' style={styles.icon} />
    <div style={styles.cardTitle}>
      {title}
    </div>
  </div>
);

Card.propTypes = {
  iconLink: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

function LogoutWarningDialog(props) {
  const handleEscapeKey = (event) => {
    if (KeyboardUtils.shouldHandleCancel(event)) {
      const { onClose } = props;
      onClose();
    }
  };

  useEffect(() => {
    document.addEventListener('keydown', handleEscapeKey, false);
    // clean up after this effect:
    return function cleanup() {
      document.removeEventListener('keydown', handleEscapeKey, false);
    };
  });

  const {
    isOpen,
    onClose,
    iconLink,
    onClickOk,
  } = props;

  const title = StringUtils.titleCase(constants.navbar.title);

  return (
    <Dialog
      size='default'
      isOpen={isOpen}
      headerChildren={`Confirm [24]7ai ${title} Log Out`}
      onClickCancel={onClose}
      onClickClose={onClose}
      onClickOk={onClickOk}
      okVisible
      cancelVisible
      okChildren={'LOG OUT'}
      cancelChildren={'CANCEL'}
      styleOverride={styles.dialog}
    >
      {<Card iconLink={iconLink} title={title} />}
      <div style={styles.text}>
        <p>
          {`Logging out will end your [24]7 ${title} session, but your single sign-on session may still be active.`}
        </p>
        <strong>
          If you have other [24]7ai tools open in other tabs or browsers, you will need to log out of each tool individually.
        </strong>
        <p>
          To protect your privacy, close all browser windows when you have logged out of all tools, especially if you are using a public computer.
        </p>
      </div>
    </Dialog>
  );
}

LogoutWarningDialog.propTypes = {
  isOpen: PropTypes.bool,
  onClose: PropTypes.func.isRequired,
  onClickOk: PropTypes.func.isRequired,
  iconLink: PropTypes.string.isRequired,
};

LogoutWarningDialog.defaultProps = {
  isOpen: false,
};

export default LogoutWarningDialog;
