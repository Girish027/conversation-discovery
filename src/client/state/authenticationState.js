import {
  fromJS, Map
} from 'immutable';
import { createSelector } from 'reselect';
import axios from 'axios'; // this module is Promise based HTTP client for the browser and node.js
import appLogger from 'utils/appLogger';
import { getUserConfig } from 'state/appState';
import constants from '../constants';
import { initializeAnalytics, logAmplitudeEvent } from '../utils/amplitudeUtils';

export const types = {
  AUTHENTICATION_COMPLETE: 'auth/AUTHENTICATION_COMPLETE',
  LOGGED_OUT: 'auth/LOGGED_OUT'
};

export const stateKey = {
  name: 'name',
  oktaUrl: 'oktaUrl',
  email: 'email'
};

const devlog = appLogger({ tag: 'authenticationState' });

export const initialAuthState = fromJS({
  name: 'User Name',
  oktaUrl: ''
});

export const actionCompletedAuth = (userInfo) => ({
  type: types.AUTHENTICATION_COMPLETE,
  userInfo
});

export const actionLoggedoutAuth = () => ({
  type: types.LOGGED_OUT
});

export const getAuthUserinfo = () => (dispatch) => {
  axios.get(constants.serverApiUrls.authentications)
    .then((res) => {
      const authInfoArray = res.data;
      if (Array.isArray(authInfoArray) && authInfoArray.length) {
        const authInfo = authInfoArray[0];
        const userInfo = {
          name: authInfo.name,
          oktaUrl: authInfo.oktaUrl,
          email: authInfo.email,
        };
        initializeAnalytics(authInfo.analyticsKey, authInfo.name);
        const IDTUserEvent = { username: authInfo.name, toolId: constants.toolName, env: constants.environment };
        logAmplitudeEvent('IDTUser', IDTUserEvent);
        dispatch(actionCompletedAuth(userInfo));
        dispatch(getUserConfig());
      } else {
        dispatch(actionLoggedoutAuth());
      }
    })
    .catch((fetchErr) => {
      devlog.error(fetchErr);
    });
};

// REDUCER
export const authReducer = (state = initialAuthState, action = {}) => {
  switch (action.type) {
    case types.AUTHENTICATION_COMPLETE:
      return state.merge(Map(action.userInfo));
    case types.LOGGED_OUT:
      return initialAuthState;
    default:
      return state;
  }
};

// SELECTORS
export const getAuthState = (state) => state.get('auth');

export const getAuthNameSelector = createSelector(
  getAuthState,
  (authState) => authState.get(stateKey.name),
);

export const getOktaUrlSelector = createSelector(
  getAuthState,
  (authState) => authState.get(stateKey.oktaUrl),
);

export const getEmailSelector = createSelector(
  getAuthState,
  (authState) => authState.get(stateKey.email),
);
