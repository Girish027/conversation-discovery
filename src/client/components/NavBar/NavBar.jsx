import React from 'react';
import PropTypes from 'prop-types';
import { NavBar as TfsNavBar, MultiLevelPicker } from '@tfs/ui-components';
import axios from 'axios';
import { fromJS } from 'immutable';
import { convertFirstLetterUppercase } from 'utils/stringOperations';
import appLogger from 'utils/appLogger';
import { getMatchingAccountId, getComponentClientIdForClient } from 'utils/caaHelper';
import {
  getAuthUserinfo
} from 'state/authenticationState';
import { routeNames, getRoute } from 'utils/RouteHelper';
import { push } from 'connected-react-router';
import {
  // registerSessionCAA,
  changeClient,
} from 'state/appState';
import constants from '../../constants';
import _ from 'lodash';
import { logAmplitudeEvent } from '../../utils/amplitudeUtils';
import LogoutWarningDialog from './LogoutWarningDialog';

const devlog = appLogger({ tag: 'NavBar' });

export default class NavBar extends TfsNavBar {
  constructor(props) {
    super(props);
    this.logoutOkta = this.logoutOkta.bind(this);
    this.onClientAppSelected = this.onClientAppSelected.bind(this);
    this.onClientPickerSelect = this.onClientPickerSelect.bind(this);
    this.productNameSelected = this.productNameSelected.bind(this);
    this.getClientAppConfigArray = this.getClientAppConfigArray.bind(this);
    this.clientsConfig = {
      title: 'Choose Workspace',
      items: [],
    };

    this.state = {
      showClientPicker: false,
      selectedClientConfig: {},
      showLogoutWarning: false,
    };
    this.displayClientName = [];
    this.selectedClientName = '';
    this.isCloseDisabled = true;
  }

  componentDidMount() {
    this.props.dispatch(getAuthUserinfo());
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.unifiedPortalClientPicker !== nextProps.unifiedPortalClientPicker) {
      const { clientId, appId } = this.props;
      this.setState({
        showClientPicker:
          nextProps.unifiedPortalClientPicker,
        selectedClientConfig: { clientId, appId },
      });
    }
  }

  getClientAppConfigArray(clientsConfig) {
    if (_.isEmpty(clientsConfig) || !clientsConfig.items.length) {
      const { clientConfigs } = this.props;
      const clientList = clientConfigs.toJS();

      if (clientList.items && clientList.items.length > 0) {
        return clientList;
      }

      if (!_.isNil(clientList) && clientList.length > 0) {
        let currentClient = clientList[0].clientId;
        const getAppsList = (accumulator = [], clientEntry) => {
          accumulator.push({
            title: clientEntry.appId,
            field: 'appId',
            data: {
              appId: clientEntry.appId,
            },
          });
          return accumulator;
        };

        clientList.forEach((client, index) => {
          const existingConfig = clientsConfig.items.find((config) => config.title === client.clientId);
          if (_.isNil(existingConfig)) {
            currentClient = client.clientId;
            const clientName = client.clientDisplayName;
            const clientEntries = clientList.slice(index).filter((cl) => cl.clientId === currentClient);
            const items = clientEntries[0].apps.reduce(getAppsList, []);
            clientsConfig.items.push({
              title: clientName,
              field: 'clientId',
              data: {
                clientId: currentClient,
              },
              items,
            });
          }
        });
      }
    }
    return clientsConfig;
  }

  logoutOkta() {
    axios(`${this.props.oktaUrl}${constants.oktaApiUrls.currentSession}`, {
      method: 'DELETE',
      withCredentials: true
    })
      .then(() => {
        const IDTUserLogOutEvent = { username: this.props.userName, toolId: constants.toolName, env: constants.environment };
        logAmplitudeEvent('IDTUserLogOutEvent', IDTUserLogOutEvent);
        window.location.href = constants.navbar.logoutUrl;
      })
      .catch((err) => {
        devlog.trace(err.response);
      });
  }

  onClientAppSelected(newClientId, newAppId) {
    const clientConfigs = this.props.clientConfigs.toJS();
    const {
      clientId, appId, componentClientId, dispatch
    } = this.props;
    let newComponentClientId = componentClientId;

    if (clientId !== newClientId || appId !== newAppId) {
      if (clientId !== newClientId || newComponentClientId === undefined || newComponentClientId === '') {
        newComponentClientId = getComponentClientIdForClient(clientConfigs, newClientId);
      }
      const newAccountId = getMatchingAccountId(clientConfigs, newClientId, newAppId);
      dispatch(changeClient({
        clientId: newClientId,
        appId: newAppId,
        accountId: newAccountId,
        componentClientId: newComponentClientId
      }));

      const newRoute = getRoute(routeNames.DISCOVER_INTENTS, {
        clientId: newClientId,
        appId: newAppId,
        accountId: newAccountId,
        componentClientId: newComponentClientId
      });
      dispatch(push(newRoute));
    }
  }

  productNameSelected() {
    const clientConfigs = this.props.clientConfigs.toJS();
    const {
      clientId, appId, componentClientId, dispatch
    } = this.props;
    const newComponentClientId = componentClientId;

    const newAccountId = getMatchingAccountId(clientConfigs, clientId, appId);

    const newRoute = getRoute(routeNames.DISCOVER_INTENTS, {
      clientId,
      appId,
      accountId: newAccountId,
      componentClientId: newComponentClientId
    });
    dispatch(push(newRoute));
  }

  onClientPickerSelect(selectedClientConfig) {
    const {
      clientId, appId, title,
    } = selectedClientConfig;

    if (title.length > 0) {
      this.displayClientName = [title[0], convertFirstLetterUppercase(title[1])];
      const [first] = title;
      this.selectedClientName = first;
    }

    if (clientId && appId) {
      this.setState({
        showClientPicker: false,
        selectedClientConfig: {
          clientId,
          appId,
        },
      }, () => {
        this.onClientAppSelected(clientId, appId);
      });
    }
  }

  getIconLink = () => {
    const { unifiedPortalUrl } = this.props;
    const iconName = `${constants.ICON_DISCOVERY}.svg`;
    return `${unifiedPortalUrl}assets/${iconName}`;
  }

  render() {
    const { clientId = ' ', appId = '', clientConfigs } = this.props;
    let { userName } = this.props;
    const { contactSupportUrl, oktaUserAccountUrl } = this.props;
    const { email } = this.props;
    const { docPortalUrl, unifiedPortalUrl } = this.props;
    const { showClientPicker, selectedClientConfig, showLogoutWarning } = this.state;
    const unifiedPortalLink = `${unifiedPortalUrl}?clientid=${clientId}`;
    const clientList = clientConfigs.toJS();
    if ((_.isEmpty(this.clientsConfig) || !this.clientsConfig.items.length) && !_.isNil(clientList) && clientList.length > 0) {
      this.clientsConfig = this.getClientAppConfigArray(this.clientsConfig);
    }

    // TODO: user name should be full name instead of generating from email.(See authentication.js)
    userName = convertFirstLetterUppercase(userName);

    if ((this.selectedClientName === clientId || this.selectedClientName === '') && clientId && appId) {
      this.selectedClientName = clientId;
      for (let i = 0; i < clientList.length; i += 1) {
        const client = clientList[i];

        if (client.clientId === clientId) {
          this.selectedClientName = client.clientDisplayName;
          break;
        }
      }
      this.displayClientName = [this.selectedClientName, convertFirstLetterUppercase(appId)];
    }

    return (
      <div>
        <MultiLevelPicker
          show={showClientPicker}
          config={this.clientsConfig}
          selection={selectedClientConfig}
          maxDepth={2}
          shouldRetainState
          shouldAutoSelect
          isCloseDisabled={this.isCloseDisabled}
          onCancel={() => this.setState({ showClientPicker: false })}
          onSelect={this.onClientPickerSelect}
        />
        { showLogoutWarning
          && (
            <LogoutWarningDialog
              isOpen
              onClose={() => {
                this.setState({ showLogoutWarning: false });
              }}
              onClickOk={this.logoutOkta}
              productName={constants.navbar.title}
              iconLink={this.getIconLink()}
            />
          )}
        <TfsNavBar
          clientValues={this.displayClientName}
          defaultClientLabel='CHOOSE APP'
          productName={constants.navbar.title}
          onClickProductName={this.productNameSelected}
          homeLink={unifiedPortalLink}
          docLink={docPortalUrl}
          supportLink={contactSupportUrl}
          userInfo={{
            name: userName,
            email,
            onSignOut: () => {
              this.setState({ showLogoutWarning: true });
            },
            accountLink: oktaUserAccountUrl
          }}
          onClickClient={() => {
            this.isCloseDisabled = false;
            this.setState({ showClientPicker: true });
          }}
        />
      </div>
    );
  }
}

NavBar.propTypes = {
  dispatch: PropTypes.func,
  userName: PropTypes.string,
  oktaUrl: PropTypes.string,
  email: PropTypes.string,
  clientId: PropTypes.string,
  componentClientId: PropTypes.string,
  appId: PropTypes.string,
  clientConfigs: PropTypes.object,
  unifiedPortalClientPicker: PropTypes.bool,
  docPortalUrl: PropTypes.string,
  unifiedPortalUrl: PropTypes.string,
};

NavBar.defaultProps = {
  userName: 'userName',
  email: 'email',
  oktaUrl: '',
  clientId: '',
  componentClientId: '',
  appId: '',
  clientConfigs: fromJS([]),
  unifiedPortalClientPicker: false,
  docPortalUrl: '',
  unifiedPortalUrl: '',
  dispatch: () => {},
};
