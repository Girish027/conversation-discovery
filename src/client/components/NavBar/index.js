import { connect } from 'react-redux';
import NavBar from './NavBar';

import {
  getAuthNameSelector,
  getOktaUrlSelector,
  getEmailSelector
} from '../../state/authenticationState';

import {
  getClientIdSelector,
  getAppIdSelector,
  getClientConfigsSelector,
  getComponentClientIdSelector,
  getUnifiedPortalClientPicker,
  getDocPortalUrlSelector,
  getUnifiedPortalUrlSelector,
  getUserTypeSelector,
  getContactSupportUrlSelector,
  getOktaUserAccountUrlSelector,
} from '../../state/appState';

const mapStateToProps = (state) => ({
  userName: getAuthNameSelector(state),
  oktaUrl: getOktaUrlSelector(state),
  email: getEmailSelector(state),
  clientId: getClientIdSelector(state),
  appId: getAppIdSelector(state),
  componentClientId: getComponentClientIdSelector(state),
  clientConfigs: getClientConfigsSelector(state),
  unifiedPortalClientPicker: getUnifiedPortalClientPicker(state),
  docPortalUrl: getDocPortalUrlSelector(state),
  unifiedPortalUrl: getUnifiedPortalUrlSelector(state),
  userType: getUserTypeSelector(state),
  contactSupportUrl: getContactSupportUrlSelector(state),
  oktaUserAccountUrl: getOktaUserAccountUrlSelector(state),
});

const mapDispatchToProps = (dispatch) => ({
  dispatch
});

export default connect(mapStateToProps, mapDispatchToProps)(NavBar);
