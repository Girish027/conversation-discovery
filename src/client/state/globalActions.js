import { downloadFile } from 'utils/FileUtils';
import { getCAASelector } from 'state/appState';
import { getActiveProjectId } from 'state/projectsState';
import Constants from '../constants';
import { logAmplitudeEvent } from '../utils/amplitudeUtils';
import {
  DOWNLOAD_FILE
} from './types';

export const downloadResults = (constructURL, runId, clusterId = '') => (dispatch, getState) => {
  const state = getState();
  const projectId = getActiveProjectId(state);
  const caa = getCAASelector(state);
  const locationUrl = constructURL({
    ...caa, projectId, runId, clusterId
  });
  dispatch({ type: DOWNLOAD_FILE, locationUrl });
  const downloadRun = { toolId: Constants.toolName, runId, env: Constants.environment };
  logAmplitudeEvent('IDT_DownloadRunResults', downloadRun);
  downloadFile(locationUrl);
};
