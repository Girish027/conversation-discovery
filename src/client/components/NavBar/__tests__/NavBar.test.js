import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import NavBar from 'components/NavBar/NavBar';
import { fromJS } from 'immutable';
import * as authState from 'state/authenticationState';
import * as appState from 'state/appState';

jest.mock('state/appState');

describe('<NavBar />', function () {
  let wrapper;

  const caa = {
    clientId: 'assurant',
    appId: 'pyb',
    accountId: 'pyb',
  };
  const clientConfigs = [{
    clientId: 'assurant',
    apps: [{
      appId: 'pyb',
      accountId: 'pyb'
    }]
  }, {
    clientId: 'capitalone',
    apps: [{
      appId: 'chatbot',
      accountId: 'chatbot'
    }]
  }, {
    clientId: 'ebay',
    apps: [{
      appId: 'uk',
      accountId: 'uk'
    }, {
      appId: 'de',
      accountId: 'de'
    }]
  }, {
    clientId: 'sirius',
    apps: [{
      appId: 'chatbot',
      accountId: 'chatbot'
    }]
  }];


  const props = {
    ...caa,
    clientConfigs: fromJS(clientConfigs),
    userName: 'abc def',
    oktaUrl: 'http://okta.com',
    dispatch: () => {},
  };

  beforeEach(function () {
    authState.getAuthUserinfo = jest.fn(() => 'getAuthUserinfo');
  });

  afterAll(() => {
    jest.clearAllMocks();
  });

  const getShallowWrapper = (propsObj) => shallow(<NavBar
    {...propsObj}
  />);

  describe('Creating an instance', () => {
    test('should exist - basic NavBar component', () => {
      wrapper = getShallowWrapper(props);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly when information is not yet available', () => {
      wrapper = getShallowWrapper();
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly when there is information', () => {
      wrapper = getShallowWrapper(props);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Functionality:', () => {
    beforeAll(() => {
      props.dispatch = jest.fn();
      appState.changeClient = jest.fn();
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    describe('onClientAppSelected', () => {
      test('should dispatch action to change client, if selected client changes', () => {
        appState.changeClient.mockReturnValue('changeClient');
        wrapper = getShallowWrapper(props);
        wrapper.instance().onClientAppSelected('ebay', 'uk');
        expect(appState.changeClient).toHaveBeenCalledWith({
          clientId: 'ebay',
          appId: 'uk',
          accountId: 'uk',
        });
        expect(props.dispatch).toHaveBeenCalledWith('changeClient');
      });

      test('should not dispatch action to change client, if the newly selected client is same', () => {
        wrapper = getShallowWrapper(props);
        wrapper.instance().onClientAppSelected('assurant', 'pyb');
        expect(props.dispatch).not.toHaveBeenCalledWith('changeClient');
      });

      test('productNameSelected - should push reload route', () => {
        wrapper = getShallowWrapper(props);
        wrapper.instance().productNameSelected();
        expect(props.dispatch).toHaveBeenCalled();
      });

      test('should change the state and update the selected client and app', () => {
        wrapper = getShallowWrapper(props);
        wrapper.instance().onClientPickerSelect({
          clientId: 'assurant',
          appId: 'pyb',
          title: '',
        });
        expect(wrapper.state().showClientPicker).toEqual(false);
        expect(wrapper.state().selectedClientConfig).toEqual({ clientId: 'assurant', appId: 'pyb' });
      });

      test('should format the client config as per new navbar', () => {
        const config = {
          title: 'Choose Workspace',
          items: [],
        };

        const clientConfig = [{
          clientId: 'assurant',
          clientDisplayName: 'componentAssurant',
          apps: [{
            appId: 'pyb',
            accountId: 'pyb'
          }]
        }];

        const newProps = {
          ...caa,
          clientConfigs: fromJS(clientConfig),
          userName: 'abc def',
          oktaUrl: 'http://okta.com',
          dispatch: () => {},
        };

        wrapper = getShallowWrapper(newProps);
        const clientList = wrapper.instance().getClientAppConfigArray(config);
        expect(clientList).toEqual({
          items: [
            {
              data: {
                clientId: 'assurant'
              },
              field: 'clientId',
              items: [
                {
                  data: {
                    appId: 'pyb'
                  },
                  field: 'appId',
                  title: 'pyb'
                }
              ],
              title: 'componentAssurant'
            }
          ],
          title: 'Choose Workspace'
        });
      });
    });
  });
});
