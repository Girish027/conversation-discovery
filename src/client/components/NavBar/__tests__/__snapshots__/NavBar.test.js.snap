// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> Snapshots renders correctly when information is not yet available 1`] = `
<div>
  <withTheme(MultiLevelPickerDialog)
    config={
      Object {
        "items": Array [],
        "title": "Choose Workspace",
      }
    }
    isCloseDisabled={true}
    maxDepth={2}
    onCancel={[Function]}
    onSelect={[Function]}
    selection={Object {}}
    shouldAutoSelect={true}
    shouldRetainState={true}
    show={false}
  />
  <withTheme(NavBar)
    clientValues={Array []}
    defaultClientLabel="CHOOSE APP"
    docLink=""
    homeLink="?clientid="
    onClickClient={[Function]}
    onClickProductName={[Function]}
    productName="INTENT DISCOVERY"
    userInfo={
      Object {
        "accountLink": undefined,
        "email": "email",
        "name": "Username",
        "onSignOut": [Function],
      }
    }
  />
</div>
`;

exports[`<NavBar /> Snapshots renders correctly when there is information 1`] = `
<div>
  <withTheme(MultiLevelPickerDialog)
    config={
      Object {
        "items": Array [
          Object {
            "data": Object {
              "clientId": "assurant",
            },
            "field": "clientId",
            "items": Array [
              Object {
                "data": Object {
                  "appId": "pyb",
                },
                "field": "appId",
                "title": "pyb",
              },
            ],
            "title": undefined,
          },
          Object {
            "data": Object {
              "clientId": "capitalone",
            },
            "field": "clientId",
            "items": Array [
              Object {
                "data": Object {
                  "appId": "chatbot",
                },
                "field": "appId",
                "title": "chatbot",
              },
            ],
            "title": undefined,
          },
          Object {
            "data": Object {
              "clientId": "ebay",
            },
            "field": "clientId",
            "items": Array [
              Object {
                "data": Object {
                  "appId": "uk",
                },
                "field": "appId",
                "title": "uk",
              },
              Object {
                "data": Object {
                  "appId": "de",
                },
                "field": "appId",
                "title": "de",
              },
            ],
            "title": undefined,
          },
          Object {
            "data": Object {
              "clientId": "sirius",
            },
            "field": "clientId",
            "items": Array [
              Object {
                "data": Object {
                  "appId": "chatbot",
                },
                "field": "appId",
                "title": "chatbot",
              },
            ],
            "title": undefined,
          },
        ],
        "title": "Choose Workspace",
      }
    }
    isCloseDisabled={true}
    maxDepth={2}
    onCancel={[Function]}
    onSelect={[Function]}
    selection={Object {}}
    shouldAutoSelect={true}
    shouldRetainState={true}
    show={false}
  />
  <withTheme(NavBar)
    clientValues={
      Array [
        undefined,
        "Pyb",
      ]
    }
    defaultClientLabel="CHOOSE APP"
    docLink=""
    homeLink="?clientid=assurant"
    onClickClient={[Function]}
    onClickProductName={[Function]}
    productName="INTENT DISCOVERY"
    userInfo={
      Object {
        "accountLink": undefined,
        "email": "email",
        "name": "Abc Def",
        "onSignOut": [Function],
      }
    }
  />
</div>
`;
